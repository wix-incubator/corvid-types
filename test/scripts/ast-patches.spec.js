const path = require("path");
const ts = require("typescript");
const changeWixAuthElevateFunctionType = require("../../scripts/ast-patches/changeWixAuthElevateFunctionType");

const fixturesDir = path.join(__dirname, "./fixtures");

describe("Ast patches", () => {
  describe("ChangeWixAuthElevateFunctionType", () => {
    it("Should apply patch", () => {
      const sourceFilePath = path.join(fixturesDir, "wix-auth.d.ts");
      const tsProgram = ts.createProgram([sourceFilePath], {});
      const sourceAst = tsProgram.getSourceFile(sourceFilePath);

      const afterPatchAst = changeWixAuthElevateFunctionType(sourceAst);
      const result = ts.createPrinter().printFile(afterPatchAst);

      expect(result).toMatchSnapshot();
    });
  });
});
