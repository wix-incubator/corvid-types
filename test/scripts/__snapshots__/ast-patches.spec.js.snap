// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ast patches ChangeWixAuthElevateFunctionType Should apply patch 1`] = `
"declare module 'wix-auth' {
    /**
     * Returns an instance of the specified function for which the site visitor's permissions have been elevated to the highest level required.
     * 	[Read more](https://www.wix.com/corvid/reference/wix-auth.html#elevate)
     */
    function elevate<T extends (...arg: any) => any>(func: T): (...param: Parameters<T>) => ReturnType<T>;
}
"
`;
